src/IPAddr.cc

```diff
@@ -33,8 +33,9 @@ ConnKey& ConnKey::operator=(const ConnKey& rhs) {
 
     // Because of padding in the object, this needs to memset to clear out
     // the extra memory used by padding. Otherwise, the session key stuff
-    // doesn't work quite right.
-    memset(this, 0, sizeof(ConnKey));
+    // doesn't work quite right. The static_cast is to silence a
+    // -Wnontrival-memcall warning from clang++ 17 and later.
+    memset(static_cast<void*>(this), 0, sizeof(ConnKey));
```

Comment:
I wonder about explicitly pushing  `-Wno-nontrivial-memcall` to ignore this, mostly worried that we copy&paste the line without the comment somewhere and move on.